### PR TITLE
Added .res-orangered-hideEmptyModmailButton to orangered css

### LINKED ISSUE #3883: Orangered options to hide envelope and modmail buttons if no mail
https://en.reddit.com/r/Enhancement/comments/5nb5in/feature_request_hide_the_old_modmail_button_if/

The new modmail is excellent and all of the subreddits that I mod have switched over to it. I assume most moderation teams have done the same. While replies to conversations from prior to the switch (>1 month ago) still go to the old modmail, this is a very rare occurrence. There's thus no need to show the old modmail button in the vast majority of situations, and it would be nice if it was hidden.

### PR DESCRIPTION
**Pull request by kevinliu6102**
Fixes #3883

### CODE REVIEW

File: lib/css/modules/_orangered.scss [hunk 1]
```diff
@@ -25,6 +25,13 @@
 	}
 }
 
+.res-orangered-hideEmptyModmailButton {
```
Comment (andytuba) [inline lib/css/modules/_orangered.scss]: Can you add an option to orangered.js to activate this rule? You can model it on the [`hideModmail` option](https://github.com/honestbleeps/Reddit-Enhancement-Suite/blob/e70d7963907dd8e2ec18db079fecf398de13cd6b/lib/modules/orangered.js#L75).

Comment (kevinliu6102) [inline lib/css/modules/_orangered.scss]: Hopefully the update works. I'm a beginner so a lot of this is actually flying over my head; it's not clear to me what you're asking for in request 1. I may have bitten off more than I can chew. Sorry about that.

Comment (kevinliu6102) [inline lib/css/modules/_orangered.scss]: Also realized #1234 was fixed already?

**Comment by andytuba** [comment]
To clarify, you should put "Fixes #1234" in the PR _description_ (not the title. change the "Addition for issue.."). and you shouldn't use literally 1234, you should use the issue number that you linked in the first place.

**Comment by kevinliu6102** [comment]
Like so? 

Is there something I'm missing in orangered.js?

**Comment by WilliamChiu** [comment]
Might want to replace "Fixes #1234" with "Fixes #3883"

File: lib/css/modules/_orangered.scss [hunk 2]
```diff
@@ -25,7 +25,7 @@
 	}
 }
 
-.res-orangered-hideEmptyModmailButton {
+.res-orangered-hideEmptyModMailButton {
 	#modmail.nohavemail,
 	#modmail.nohavemail + .separator {
 	    display: none;
```
Comment (WilliamChiu) [inline lib/css/modules/_orangered.scss]: Before 'display: none;', use a tab instead of four spaces in order to pass the lint check

Comment (kevinliu6102) [inline lib/css/modules/_orangered.scss]: Thanks for the advice.

**Comment by andytuba** [comment]
@kevinliu6102 👍  for the option you added to orangered.js.

It would also be good to add two more options / CSS rules which do the same thing for the main envelope icon `a#mail` and the new modmail icon. `a#new_modmail`

**Comment by kevinliu6102** [comment]
@andytuba I think this should do it?

**Comment by andytuba** [comment]
@kevinliu6102 Yay! Just one more thing -- adding the titles/descriptions (e.g. `orangeredHideEmptyModMailDesc `) into `en.json`.

It would also be nice to order the options "mail, modmail, newmodmail".

File: lib/css/modules/_orangered.scss [hunk 3]
```diff
@@ -25,6 +25,27 @@
...
+	}
+}
+
+.res-orangered-hideEmptyNewModMail {
+	#new_modmail.nohavemail,
+	#new_modmail.nohavemail + .separator {
+		display: none;
+	}
+}
+
```
Comment (larsjohnsen) [inline lib/css/modules/_orangered.scss]: These may be more readable as

```
.res-orangered-hideEmptyMail #mail,
.res-orangered-hideEmptyModMail #modmail,
.res-orangered-hideEmptyNewModMail #new_modmail {
	&.nohavemail,
	&.nohavemail + .separator {
		display: none;
	}
}

**Comment by kevinliu6102** [comment]
@andytuba I actually went ahead and added a general option for hiding the new modmail button since I saw the original modmail button had such an option. If it's not needed, just let me know and I'll get rid of it.
@larsjohnsen Thanks for the advice.

**Pull request merged**